Add first draft maintainers blog for April 2025 (#3329)
nf-core/website: sites/main-site/src/content/blog/2025/maintainers-minutes-2025-04-24.mdx

---
title: "Maintainers Minutes: April 2025"
subtitle: "Keeping you informed of the latest maintainers discussions"

> pubDate: 2025-04-29T18:00:00+01:00

Author: **suggestion:** 
```suggestion
pubDate: 2025-05-02T10:00:00+01:00
```

<Profile username="atrigila">Anabella Trigila</Profile>
<Profile username="LouisLeNezet">Louis Le Nézet</Profile>
<Profile username="pontus">Pontus Freyhult</Profile>

## Use of CODEOWNER files

<Image
    src={notifications}
    class="d-block m-auto"
    width={400}
    density={[1.5, 2]}
    alt="Captain Jack Sparrow from Pirates of the Caribbean running away from a large crowd of people with the texts 'being added to the nf-core/modules codeowners file' and 'notifications' overlaid on Captain Jack and the crowd respectively."
/>

Maxime brought up the use of GitHub's [`CODEOWNER`](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) files.
These files are used to automate 🛎️ notifications to certain people listed in the file to ensure 'the right people' are informed to look at pull requests.

This had previously been used within nf-core/modules, but was [eventually removed](https://github.com/nf-core/modules/pull/6738) due to complaints from prolific module creators getting inundated by notifications on update PRs of modules they were no longer interested in maintaining themselves.

> However Maxime and Edmund proposed resurrecting the use of these files pipelines (possibly with opt-in).

Reviewer: ```suggestion
However Maxime and Edmund proposed resurrecting the use of these files in pipelines (possibly with opt-in).
```

There were still a few concerns that possible some developers will complain of getting too many notifications on their pipelines.

We therefore decided that Maxime would ✒️ write a more formal proposal document, in the meantime volunteers will manually add the file to their pipelines (to see if their co-developers complain about notifications), and check the results in 2 months.

## Use of nf-core/modules issues as wishlists

<Image
    src={botoverlords}
    class="d-block m-auto"
    width={400}
    density={[1.5, 2]}
    alt="Screenshot of a news presenter from the cartoon The Simpsons with the text 'I for one one welcome our robot overlords'"
/>

Famke raised an issue she and the nf-core/modules sub-team encountered during the spring cleaning.
They found that there was a 📈 huge number of open issues on the repository requesting new modules, some very old, that were never assigned to anyone nor worked on.
The large number made it very hard to go through, check the status of the various issues, and generally reasonably maintain a sense of order.

> The sub-teams impression was that many people were using the issues page as 'wish list', where someone wants a module integrated into a pipeline, but didn't want to do it themselves (either due to experience or time).

Reviewer: ```suggestion
The sub-team impression was that many people were using the issues page as 'wish list', where someone wants a module integrated into a pipeline, but didn't want to do it themselves (either due to experience or time).
```

To reduce the maintenance burden for the maintainers team, we discussed a couple of options, including having a singular mega-issue where all module 'requests' were recorded, or only allowing a new-module issue to be created if the person was willing to work on it themselves.
James' concern was that the list of 'unassigned' modules can sometimes be very useful for training on writing modules or for hackathons where novice participants don't have something specific to work on (or want to practice before working on a specific pipeline).
Removing or 'hiding' these issues would mean these would potentially hinder teachers or hackathon newcomers effectively begin their contributions.

Eventually we found a compromise via a 🤖 'stale bot' concept that was assigned to Famke to implement.
New-module issues that are not immediately assigned to work on by someone will be automatically given a 'wishlist' label.

> If it remains unassigned after a period time (initially around a year to ensure we always have a list of 'free' modules for hackathons), the issue wil be closed by an automated bot.

Reviewer 2: ```suggestion
If it remains unassigned after a period time (initially around a year to ensure we always have a list of 'free' modules for hackathons), the issue will be closed by an automated bot.
```

The time period would reset if assigned to someone and then subsequently unassigned (if they could not complete it).

Through this automated mechanism we hope it will make the nf-core/modules (our most popular repository!) more manageable for the community and the maintainers team, freeing up time to work on cooler things (or argue about naming things..., the maintainer team's favourite pasttime 😉).

## Saving more polar bears via optimised AWS megatest configs

<Image

>     src={notifications}

Reviewer 3: ```suggestion
    src={saveapolarbear}
```

Reviewer 3: Currently using wrong image for megatest section!

    class="d-block m-auto"
    width={400}
    density={[1.5, 2]}
    alt="Photo of a polar bear face with the text 'Save a polarbear, optimise your AWS megatests' overlaid"
/>

Finally, we briefly looked at a neat [little repository](https://github.com/FloWuenne/megatest-resource-optimization/tree/main/optimized_configs) that Florian (who unfortunately sent his apologies) has made that includes optimised resource-usage configuration files for nf-core pipelines AWS megatests.

> This was after observations that many pipeline's megatest configurations could be rather wasteful in over- or under-requesting resources for AWS - resulting in wasted costs and energy by using larger spot instances than necessary or causing unnecessary retries (sad 🐻‍❄️).

Reviewer 3: ```suggestion
This was after observations that many pipeline megatest configurations could be rather wasteful in over- or under-requesting resources on AWS (and in general), resulting in wasted costs and energy by requesting more cpus and memory than their processes actually use (sad 🐻‍❄️).
```

We discussed whether there should be an automation to generate an optimised config and updating a pipeline's `test_full` config with the optimised config, or some other mechanism to incorporate these into our pipelines.

> We decided to ask volunteers to implement Florian's existing optimised configs to see if they cause any problems on a subset of nf-core pipelines before rolling out, and I discuss further on Slack (in fact, the Slack discussions were already fruitful and @Maxime and @Florian have already come up with a potential non-invasive system without requiring any extra work by pipeline developers - watch this space!)

Reviewer 2: ```suggestion
We decided to ask volunteers to implement Florian's existing optimised configs to see if they cause any problems on a subset of nf-core pipelines before rolling out. The Slack discussions were already fruitful. @Maxime and @Florian already came up with a potential non-invasive system without requiring any extra work by pipeline developers - watch this space!
```

Reviewer 2: I didnt see the slack discussion, but add yourself as well if you were part of it @jfy133. I didn't quite if you were also involved in the discussion

Author: The discussion was here: https://nfcore.slack.com/archives/C043UU89KKQ/p1745593967346769?thread_ts=1745246825.260859&cid=C043UU89KKQ

I basically just said 'HELL YEAH' to the proposal from Maxime and Florian, so I don't think I'm necessary here

> We decided to ask volunteers to implement Florian's existing optimised configs to see if they cause any problems on a subset of nf-core pipelines before rolling out, and I discuss further on Slack (in fact, the Slack discussions were already fruitful and @Maxime and @Florian have already come up with a potential non-invasive system without requiring any extra work by pipeline developers - watch this space!)

Reviewer: we should add a link to the slack channel/thread or at least the name of the channel

Reviewer: ah ok! then I would mention it was "internal discussion", otherwise I would expect to be able to chime in 😄

Author: Updated accoerdingly!

> We decided to ask volunteers to implement Florian's existing optimised configs to see if they cause any problems on a subset of nf-core pipelines before rolling out, and I discuss further on Slack (in fact, the Slack discussions were already fruitful and @Maxime and @Florian have already come up with a potential non-invasive system without requiring any extra work by pipeline developers - watch this space!)

Author: **suggestion:** 
```suggestion
We decided to ask volunteers to implement Florian's existing optimised configs to see if they cause any problems on a subset of nf-core pipelines before rolling out, and I discuss further on Slack (in fact, the internal maintainers team Slack discussions were already fruitful and @Maxime and @Florian have already come up with a potential non-invasive system without requiring any extra work by pipeline developers - watch this space!)
```

## Misc

Finally a couple of minor other points were briefly announced in the last couple of minutes:

- James asked for 'live reviews' of his PR to the [test-data specifications](https://nf-co.re/docs/guidelines/components/test_data#replication-of-test-data) to improve community member's use of nf-core/testdatasets
    - TL;DR: module tests don't need to make sense, the tool just needs to 'work'! Re-use existing test-data as much as possible.

> - Júlia briefly summarised an upcoming small patch release for nf-core/tools to fix a couple of lingering bugs that have cropped up in the last couple of weeks related to AWS megatests and Nextflow itself.

Reviewer: ```suggestion
- Júlia briefly summarised an upcoming small patch release for nf-core/tools to fix a couple of CI bugs that have cropped up in the last couple of weeks related to AWS megatests and Nextflow itself.
```

## The end

> As always, if you want to get involved and give your input, join the discussion on relevant PRs and slack threads!

Reviewer 2: ```suggestion
As always, if you want to get involved and give your input, join the discussion on relevant PRs and Slack threads!
```
